### ISSUE TITLE
pgsql: Insert_ID() - PostgreSQL8 driver assumes that lastval() will be set.

### ISSUE THREAD

**Issue by NathanGibbs3** [ADOdb/ADOdb#978]
### Description
Getting the error and stack trace below in CI when calling `Insert_ID()` on a newly initialized ADOdb object connected to a PostgreSQL DB.
~~~
1) dbTest::testbaseInsertIDReturnsExpected
postgres9 error: [-1: ERROR:  lastval is not yet defined in this session] in EXECUTE("SELECT lastval()")

/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/adodb-errorhandler.inc.php:84
/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/adodb.inc.php:1624
/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/adodb.inc.php:1590
/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/adodb.inc.php:2142
/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/drivers/adodb-postgres8.inc.php:51
/home/travis/build/NathanGibbs3/BASE/build/adodb/ADOdb-5.22.5/adodb.inc.php:1741
/home/travis/build/NathanGibbs3/BASE/includes/base_db.inc.php:467
/home/travis/build/NathanGibbs3/BASE/tests/PhpUnit/idbTest.php:353
~~~

### Environment
- ADOdb version: 5.20.0 - 5.22.5
- Driver or Module: postgres
- Database type and version: PostgreSQL 9.2 - 9.6
- PHP version: 7.0.25 - 8.1.12
- Platform: Linux

* [x] I have tested that the problem is reproducible in the [latest release](https://github.com/ADOdb/ADOdb/releases/latest)

### Steps to reproduce
Detailed, step-by-step instructions to reproduce the behavior, including:
- Initialize an ADOdb object connected to a PostgreSQL DB.
- Call `Insert_ID()`.

### Expected behavior
Would expect `Insert_ID()` to return false as per the [documentation](https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:insert_id) and not throw an error.

### Additional context 
Link to CI build were we can see the breakage on PostgreSQL for PHP versions using ADOdb 5.20+
https://app.travis-ci.com/github/NathanGibbs3/BASE/builds/263709789

**Comment by NathanGibbs3** [comment]
Link to CI build with this fixed.
https://app.travis-ci.com/github/NathanGibbs3/BASE/builds/263713300
PHP 7+ is using my ADOdb fork with this fixed.
Will PR it shortly.

FYI, something between 5.22.5 & present broke the return from calls to `metacolumns()` in the mysqli driver.

**Comment by dregad** [comment]
I just tried this and it seems to return false as expected - am I missing something ? 
 
```php
include 'adodb.inc.php'; 
$db = ADONewConnection('pgsql');
$db->connect(...);
var_dump($db->insert_id()); // false
```

**Comment by dregad** [comment]
Ah I get it - the problem occurs when using exceptions, i.e. with `include 'adodb-exceptions.inc.php';` I get

Fatal error: Uncaught ADODB_Exception: postgres9 error: [-1: ERROR:  lastval is not yet defined in this session] in EXECUTE("SELECT lastval()") in /Users/dregad/dev/adodb/adodb-exceptions.inc.php on line 97

which matches the error reported in your CI.

I'll have a look at your PR.

**Comment by NathanGibbs3** [comment]
Is it possible to backport this fix into the 5.21 and/or the 5.20 lines.
The code responsible for this issue was introduced in ADOdb 5.19

At the moment, it may be necessary to downgrade my project's maximum supported ADOdb versions to 5.18 on older PHPs.
https://github.com/NathanGibbs3/BASE/wiki/ADOdb-version-requirements-by-PHP-Version

I CI test against the minimum supported versions, If I had tested against the maximum supported versions, this issue would have been present in the PHP 5 tests as well.

Also, I may have to take into account users running ADOdb > 5.18 on PHP 5x and compensate for this error in my project's code, as the currently usable versions of ADOdb for those versions of PHP do not.

**Comment by NathanGibbs3** [comment]
I have PR #983 for the 5.21 codebase.

I have https://github.com/NathanGibbs3/ADOdb/tree/NathanGibbs3-Issue%23978for5.20 in my frok for the 5.20 codebase.
Fro the 5.20 codebase
- I checked out the commit tagged v5.20.21
- Created a hotfix/5.20 branch in my fork.
- Created the https://github.com/NathanGibbs3/ADOdb/tree/NathanGibbs3-Issue%23978for5.20 branch from there.

Hopefully that will avoid adding any 5.21+ code into it, if you choose to use it.

**Comment by NathanGibbs3** [comment]
> Ah I get it - the problem occurs when using exceptions, i.e. with `include 'adodb-exceptions.inc.php';` I get

For what it's worth, I'm nor including the exception handler, just the error handler.
`include 'adodb-errorhandler.inc.php' #981 may still exist there.

**Comment by NathanGibbs3** [comment]
ADOdb 5.22x #984
ADOdb 5.21x #985
ADOdb 5.20.x https://github.com/NathanGibbs3/ADOdb/tree/NathanGibbs3-Issue%23981for5.20

**Comment by NathanGibbs3** [comment]
I understand.
As direct result of our conversation about open source development last week I went from my original position of.

> Is it possible to backport ...

to the position of
Doing the actual work & submitting PR's containing what I was originally asking you for.

If you want them in the future., they are there.
Either way, I am happy to have provided them. :smile:
That is the idea & values behind Open Source after all.

As I've stated and in alignment with your suggestion.
I'll take into account users running ADOdb 5.18 - 5.22.6 and compensate for this error in my project's code. 
I would have to do that anyways, as end users don't always update to the latest ADOdb.

Thank you again for your consideration .
